android-components/components/feature/autofill/src/main/java/mozilla/components/feature/autofill/response/dataset/LoginDatasetBuilder.kt

```diff
@@ -178,9 +180,24 @@ internal fun Dataset.Builder.setValue(
     presentation: RemoteViews,
     inlinePresentation: InlinePresentation? = null,
 ): Dataset.Builder {
-    return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R && inlinePresentation != null) {
+    return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+        val fieldBuilder: Field.Builder = Field.Builder()
+        if (value != null) {
+            fieldBuilder.setValue(value)
+        }
+        val presentationsBuilder = Presentations.Builder()
+        presentationsBuilder.setMenuPresentation(presentation)
+
+        if (inlinePresentation != null) {
+            presentationsBuilder.setInlinePresentation(inlinePresentation)
+        }
+
+        fieldBuilder.setPresentations(presentationsBuilder.build())
+        this.setField(id, fieldBuilder.build())
+    } else if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
         this.setValue(id, value, presentation, inlinePresentation)
```

Comment:
No, it should not be recursive. This actually calls `setValue` from `Dataset.Builder`. I will remove `this` keyword to avoid further confusions. Thank you for raising this!